expression/comparisons.go

```diff
@@ -41,155 +41,248 @@ func OperationNot(ai interface{}) (interface{}, error) {
+		func(a, b bool) bool {
+			return false
+		},
+		func(a, b int64) bool {
+			return a < b
+		},
 		func(a, b float64) bool {
 			return a < b
 		},
 		func(a, b string) bool {
 			return a < b
 		},
 	)
-	if err != nil {
-		return false, errors.Wrap(err, "LessThan Error")
-	}
-	return result, nil
 }
 
-func OperationLessThanEquals(ai interface{}, bi interface{}) (bool, error) {
-	result, err := ComparisonOperation(
-		ai,
-		bi,
+func OperationLessThanEquals(ai, bi interface{}) (bool, error) {
+	return ComparisonOperation(ai, bi,
+		func(a, b bool) bool {
+			return false
+		},
+		func(a, b int64) bool {
+			return a <= b
+		},
 		func(a, b float64) bool {
 			return a <= b
 		},
 		func(a, b string) bool {
 			return a <= b
 		},
 	)
-	if err != nil {
-		return false, errors.Wrap(err, "LessThanEquals Error")
-	}
-	return result, nil
 }
 
-func OperationEquals(ai interface{}, bi interface{}) (bool, error) {
-	result, err := ComparisonOperation(
-		ai,
-		bi,
+func OperationEquals(ai, bi interface{}) (bool, error) {
+	return ComparisonOperation(ai, bi,
+		func(a, b bool) bool {
+			return a == b
+		},
+		func(a, b int64) bool {
+			return a == b
+		},
 		func(a, b float64) bool {
 			return a == b
 		},
 		func(a, b string) bool {
 			return a == b
 		},
 	)
-	if err != nil {
-		return false, errors.Wrap(err, "Equals Error")
-	}
-	return result, nil
 }
 
-func OperationNotEquals(ai interface{}, bi interface{}) (bool, error) {
-	result, err := ComparisonOperation(
-		ai,
-		bi,
+func OperationNotEquals(ai, bi interface{}) (bool, error) {
+	return ComparisonOperation(ai, bi,
+		func(a, b bool) bool {
+			return a != b
+		},
+		func(a, b int64) bool {
+			return a != b
+		},
 		func(a, b float64) bool {
 			return a != b
 		},
 		func(a, b string) bool {
 			return a != b
 		},
 	)
-	if err != nil {
-		return false, errors.Wrap(err, "NotEquals Error")
-	}
-	return result, nil
 }
 
 func OperationIs(ai interface{}, bi interface{}) (bool, error) {
-	result, err := ComparisonOperation(
-		ai,
-		bi,
+	return ComparisonOperation(ai, bi,
+		func(a, b bool) bool {
+			return a == b
+		},
+		func(a, b int64) bool {
+			return a == b
+		},
 		func(a, b float64) bool {
 			return a == b
 		},
 		func(a, b string) bool {
 			return a == b
 		},
 	)
-	if err != nil {
-		return false, errors.Wrap(err, "Is Error")
-	}
-	return result, nil
 }
 
-func ComparisonOperation(ai interface{}, bi interface{}, f func(a, b float64) bool, fs func(a, b string) bool) (bool, error) {
-	switch a := ai.(type) {
-	case int64:
-		switch b := bi.(type) {
+func ComparisonOperation(
+	ai, bi interface{},
+	fbool func(a, b bool) bool,
+	fint func(a, b int64) bool,
+	ffloat func(a, b float64) bool,
+	fstr func(a, b string) bool) (bool, error) {
+
+	if ai != nil {
+		switch a := ai.(type) {
+		case bool:
+			b, ok := bi.(bool)
+			if !ok {
+				if bi != nil {
+					return false, errors.Wrapf(IncompatibleTypeError, "(bool) %v incompatible with %v", ai, bi)
+				}
+				b = false
```

Comment:
If we set these default values, should we also log or send an error somewhere so we know that we tried to compare 2 incomparable objects?